File: src/HAPM.Domain/Enums/Enums.cs
namespace HAPM.Domain.Enums;

public enum UserRole
{
    Admin = 1,
    Doctor = 2,
    Patient = 3,
    Receptionist = 4
}

public enum Gender
{
    Male = 1,
    Female = 2,
    Other = 3
}

public enum AppointmentStatus
{
    Pending = 1,
    Confirmed = 2,
    CheckedIn = 3,
    Completed = 4,
    Cancelled = 5,
    NoShow = 6
}

public enum InvoiceStatus
{
    Pending = 1,
    Paid = 2,
    Cancelled = 3,
    PartiallyPaid = 4
}

public enum WaitlistStatus
{
    Active = 1,
    Notified = 2,
    Cancelled = 3
}

public enum AuditAction
{
    Created = 1,
    Updated = 2,
    Deleted = 3
}

public enum PaymentMethod
{
    Cash = 1,
    Card = 2,
    Upi = 3,
    Insurance = 4,
    BankTransfer = 5
}

public enum LabReportStatus
{
    Uploaded = 1,
    Reviewed = 2
}

public enum NotificationType
{
    General = 0,
    AppointmentBooked = 1,
    AppointmentConfirmed = 2,
    AppointmentCancelled = 3,
    AppointmentReminder = 4,
    PrescriptionIssued = 5,
    LabReportUploaded = 6,
    InvoiceGenerated = 7,
    WaitlistSlotOpened = 8,
    PaymentReceived = 9,
    FollowUpDue = 10,
    AppointmentCompleted = 11
}

public enum StaffMessageTarget
{
    DoctorRoom = 1,
    StaffBroadcast = 2
}

File: frontend/hapm-web/src/app/shared/models/enums.ts
export type AppointmentStatus =
  | 'Pending'
  | 'Confirmed'
  | 'CheckedIn'
  | 'Completed'
  | 'Cancelled'
  | 'NoShow';

export type InvoiceStatus = 'Pending' | 'Paid' | 'Cancelled' | 'PartiallyPaid';

export type StatusTone = 'default' | 'success' | 'warning' | 'info' | 'destructive' | 'secondary';

export const APPOINTMENT_STATUS_TONE: Record<AppointmentStatus, StatusTone> = {
  Pending: 'warning',
  Confirmed: 'info',
  CheckedIn: 'secondary',
  Completed: 'success',
  Cancelled: 'destructive',
  NoShow: 'destructive',
};

export const INVOICE_STATUS_TONE: Record<InvoiceStatus, StatusTone> = {
  Pending: 'warning',
  PartiallyPaid: 'info',
  Paid: 'success',
  Cancelled: 'destructive',
};

export type WaitlistStatus = 'Active' | 'Notified' | 'Cancelled';

export const WAITLIST_STATUS_TONE: Record<WaitlistStatus, StatusTone> = {
  Active: 'info',
  Notified: 'success',
  Cancelled: 'destructive',
};

export type LabReportStatus = 'Uploaded' | 'Reviewed';

export const LAB_REPORT_STATUS_TONE: Record<LabReportStatus, StatusTone> = {
  Uploaded: 'warning',
  Reviewed: 'success',
};

export type PaymentMethod = 'Cash' | 'Card' | 'Upi' | 'Insurance' | 'BankTransfer';

export type NotificationType =
  | 'General'
  | 'AppointmentBooked'
  | 'AppointmentConfirmed'
  | 'AppointmentCancelled'
  | 'AppointmentReminder'
  | 'PrescriptionIssued'
  | 'LabReportUploaded'
  | 'InvoiceGenerated'
  | 'WaitlistSlotOpened'
  | 'PaymentReceived'
  | 'FollowUpDue'
  | 'AppointmentCompleted';

export type AuditAction = 'Created' | 'Updated' | 'Deleted';

export const AUDIT_ACTION_TONE: Record<AuditAction, StatusTone> = {
  Created: 'success',
  Updated: 'info',
  Deleted: 'destructive',
};
